dropshot/src/test_util.rs

```diff
@@ -613,6 +613,25 @@ pub async fn objects_post<S: Serialize + Debug, T: DeserializeOwned>(
     read_json::<T>(&mut response).await
 }
 
+/**
+ * Issues an HTTP DELETE to the specified object URL to delete an object.
+ */
+pub async fn object_delete<T: DeserializeOwned>(
+    client: &ClientTestContext,
+    object_url: &str,
+) -> T {
+    let mut response = client
+        .make_request_with_body(
+            Method::DELETE,
+            &object_url,
+            "".into(),
+            StatusCode::OK,
```

Comment:
Running this branch locally, I'm seeing a test failure here because the response is `StatusCode::NO_CONTENT`, not `StatusCode::OK`.

Here's where `HttpResponseDeleted` is defined: 
https://github.com/oxidecomputer/dropshot/blob/f99a003271f41f933172e33c3cc88a2f67bbdc7c/dropshot/src/handler.rs#L1234-L1251

This makes sense to me - it looks like our `HttpResponseDeleted` sends a `NO_CONTENT` response. Should that be the response we expect here?

Reply:
I think "204 No Content" is more customary for DELETE operations, especially if you're not returning anything useful in the body.  [Here's a source.](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/DELETE).